src/payment/strategies/quadpay/quadpay-payment-strategy.ts

```diff
@@ -52,23 +52,21 @@ export default class QuadpayPaymentStrategy implements PaymentStrategy {
 
         const paymentPayload = {
             methodId,
-            paymentData: { nonce },
+            paymentData: { nonce: nonce.id },
         };
 
         const { isStoreCreditApplied: useStoreCredit } = this._store.getState().checkout.getCheckoutOrThrow();
 
         await this._store.dispatch(this._storeCreditActionCreator.applyStoreCredit(useStoreCredit));
         await this._store.dispatch(this._remoteCheckoutActionCreator.initializePayment(methodId, { useStoreCredit }));
         await this._store.dispatch(this._orderActionCreator.submitOrder(order, options));
-        await this._prepareForReferredRegistration(methodId, nonce);
+        await this._prepareForReferredRegistration(methodId, nonce.id);
 
         try {
             return await this._store.dispatch(this._paymentActionCreator.submitPayment(paymentPayload));
         } catch (error) {
             if (error instanceof RequestError && error.body.status === 'additional_action_required') {
```

Comment:
Do we need to check here if nonce has a uri key ?

Reply:
Yes, I could add that validation